File: SumOfPositiveAndNegativeNumbers.cs
// Sum of Positive and Negative Numbers
// - Given an array of integers, calculate the sum of positive numbers and the sum of negative numbers separately.
// - Iterate through the array, adding each number to the respective sum based on its sign.
// - Time complexity: O(n)
// - Example: [-3, 5, -2, 7, 0, -1, 4] → Output: Positive Sum: 16, Negative Sum: -6

using System;

class Program
{
    static void Main()
    {
        int[] numbers = { -3, 5, -2, 7, 0, -1, 4 };
        int positiveSum = 0;
        int negativeSum = 0;

        for (int i = 0; i < numbers.Length; i++)
        {
            if (numbers[i] > 0)
                positiveSum += numbers[i];
            else if (numbers[i] < 0)
                negativeSum += numbers[i];
        }

        Console.WriteLine("Sum of positive numbers: " + positiveSum);
        Console.WriteLine("Sum of negative numbers: " + negativeSum);
    }
}

File: SumOfPositiveAndNegativeNumbers.py
# Sum of Positive and Negative Numbers
# - Given a list of integers, calculate the sum of positive numbers
#   and the sum of negative numbers separately.

from typing import List, Tuple

def sum_positive_negative(nums: List[int]) -> Tuple[int, int]:
    positive_sum = 0
    negative_sum = 0

    for num in nums:
        if num > 0:
            positive_sum += num
        elif num < 0:
            negative_sum += num

    return positive_sum, negative_sum


